nin/frontend/app/scripts/controllers/main.js

```diff
@@ -78,7 +78,8 @@ class MainCtrl {
             try {
               let node = demo.nm.createNode(nodeInfo);
               demo.nm.insertOrReplaceNode(node);
-            } catch (e) {
+            } catch(e) {
+              console.error(e);
```

Comment:
We should, yes. Currently the code is a spaghetti ball of yarn, and needs a refactor to be able to properly differentiate between the two cases.